kafka-rest/src/main/java/io/confluent/kafkarest/DefaultKafkaRestContext.java

```diff
@@ -54,7 +54,7 @@ public KafkaRestConfig getConfig() {
   }
 
   @Override
-  public ProducerPool getProducerPool() {
+  public synchronized ProducerPool getProducerPool() {
```

Comment:
We should use this pattern instead:

https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom

Reply:
Only problem with that approach is that client Pools may only be a singleton for this use case. Our extensions for example create a KafkaRestContext per authenticated user in order propagate identities from the HTTP request to Kafka.